### PR TITLE
unit.j: add 12 unit item/inventory functions, bug in SetUnitColor

### LINKED ISSUE #34: UnitItemInSlot
- Is it 0-indexed?
- Does it return null when there is no such item?
- What's the behavior for unit with no inventory/no inventory slot at the requested index?

### PR DESCRIPTION
**Pull request by Luashine**
Described the following functions:
UnitItemInSlot, UnitHasItem, UnitInventorySize, UnitRemoveItem,
UnitDropItemPoint, UnitDropItemTarget
UnitAddItem, UnitAddItemById, UnitAddItemToSlotById
UnitUseItem, UnitUseItemPoint, UnitUseItemTarget

Added info about a visual bug with SetUnitColor

---

Fixes #34. I think now the entire unit's item usage API is documented :tada:

### CODE REVIEW

File: unit.j [hunk 1]
```diff
@@ -514,44 +520,258 @@ native GetUnitPointValueByType takes integer unitType returns integer
...
 native UnitAddItem takes unit whichUnit, item whichItem returns boolean
 
+/**
+Creates a new item of type `itemId` and puts it in unit's inventory.
+If the inventory is full, it is dropped on the ground at unit's position instead.
+
+Returns:
+
+- item handle if the item was successfully placed in unit's inventory
+- null if inventory is full, unit cannot carry items, itemId/unit invalid etc.
```
Comment (lep) [inline unit.j:543]: So the item is still created but the native returns null? Seems a bit contradictory to the text above.

File: unit.j [hunk 2]
```diff
@@ -514,44 +520,258 @@ native GetUnitPointValueByType takes integer unitType returns integer
...
+
+@param whichUnit Target unit
+*/
 native UnitInventorySize takes unit whichUnit returns integer
 
 
+/**
+Issues an immediate order for the unit to go to point (x,y) and drop the item there.
+
+If the unit cannot reach the point, he will run up to closest location and stop there,
```
Comment (lep) [inline unit.j:628]: s/he/it

File: unit.j [hunk 3]
```diff
@@ -514,44 +520,258 @@ native GetUnitPointValueByType takes integer unitType returns integer
...
+@param slot Move to this slot
+*/
 native UnitDropItemSlot takes unit whichUnit, item whichItem, integer slot returns boolean
 
+/**
+Issues an immediate order for the `whichUnit` to go to `target` (usually another unit)
+and give the item to target. If the target has no inventory slots/full inventory,
+the item is dropped at target's feet.
+
+If the `whichUnit` cannot reach the target, he will run up to closest location and stop there,
```
Comment (lep) [inline unit.j:667]: s/he/it
Or "the unit" if you want.

File: unit.j [hunk 4]
```diff
@@ -514,44 +520,258 @@ native GetUnitPointValueByType takes integer unitType returns integer
...
+Returns:
+
+- true if item was successfully used / an order was issued
+- false otherwise, because an order was not issued:
+unit doesn't have item, item on cooldown, invalid unit/item etc.
+
+Examples:
+
+- Potion of Healing 'phea':
+   - Unit on patrol, but has full HP: does nothing, unit continues running
```
Comment (lep) [inline unit.j:702]: For a nested list indent the list item by four spaces/one tab. This is one space too few.
(I know it's a bit frustrating doing this non-interactively over comments)

File: unit.j [hunk 5]
```diff
@@ -514,44 +520,258 @@ native GetUnitPointValueByType takes integer unitType returns integer
...
+
+Returns:
+
+- true if item was successfully used / an order was issued
+- false otherwise, because an order was not issued:
+unit doesn't have item, item on cooldown, invalid unit/item etc.
+
+Examples:
+
+- Potion of Healing 'phea':
```
Comment (lep) [inline unit.j:701]: With "just" numbers it's ok to not put them in backticks but i think for more elaborate jass syntax like these fourcc literals it is nice to put them in backticks for nice syntax highlighting.

File: unit.j (hunk 1 above)
Comment (Luashine) [reply unit.j:543]: Yes it really returns null but creates an item. I will rephrase it

Comment (lep) [reply unit.j:543]: Damn that's crazy!

Comment (Luashine) [reply unit.j:543]: Added another paragraph instead of cluttering the Returns: paragraph. What do you think?

File: unit.j (hunk 2 above)
Comment (Luashine) [reply unit.j:628]: I was testing with foot*men* :laughing: Replacing with `it`

File: unit.j (hunk 5 above)
Comment (Luashine) [reply unit.j:701]: Done

File: unit.j (hunk 4 above)
Comment (Luashine) [reply unit.j:702]: Done 4 spaces...

offtopic: I said that markdown is my nemesis! I read the basic and extended syntax guides and yes this is the result :(

File: unit.j (hunk 1 above)
Comment (lep) [reply unit.j:549]: I think i'm still learning stuff from your contributions!

**Pull request merged**